diff --git a/celt/float_cast.h b/celt/float_cast.h
index 2fb87e3..14fd5b1 100644
--- a/celt/float_cast.h
+++ b/celt/float_cast.h
@@ -66,7 +66,8 @@ static OPUS_INLINE opus_int32 float2int(float x) {return _mm_cvt_ss2si(_mm_set_
 
 #include <xmmintrin.h>
 static OPUS_INLINE opus_int32 float2int(float x) {return _mm_cvt_ss2si(_mm_set_ss(x));}
 
-#elif (defined(_MSC_VER) && _MSC_VER >= 1400) && (defined(_M_X64) || (defined(_M_IX86_FP) && _M_IX86_FP >= 1))
+#elif (defined(_MSC_VER) && _MSC_VER >= 1400) && !defined(_M_ARM64EC) && !defined(_ARM64EC_) \
+   && (defined(_M_X64) || (defined(_M_IX86_FP) && _M_IX86_FP >= 1))
 
         #include <xmmintrin.h>
         static OPUS_INLINE opus_int32 float2int(float value)
